src/Core/src/Hosting/ImageSources/ImageSourceServiceProvider.cs

```diff
@@ -22,40 +25,59 @@ public ImageSourceServiceProvider(IImageSourceServiceCollection collection, ISer
 
 		public IServiceProvider HostServiceProvider { get; }
 
-		public IImageSourceService? GetImageSourceService(Type imageSource) =>
+		public IImageSourceService? GetImageSourceService([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces)] Type imageSource) =>
 			(IImageSourceService?)GetService(GetImageSourceServiceType(imageSource));
 
-		public Type GetImageSourceServiceType(Type imageSource) =>
-			_serviceCache.GetOrAdd(imageSource, type =>
-			{
-				var genericConcreteType = ImageSourceServiceType.MakeGenericType(type);
+		[UnconditionalSuppressMessage("Trimming", "IL2073", Justification = TrimmerHelper.ConcurrentDictionary)]
+		[UnconditionalSuppressMessage("Trimming", "IL2111", Justification = TrimmerHelper.MakeGenericType)]
```

Comment:
Why is this needed? There's no MakeGenericType call in this method.

Reply:
The `IL2111` came from `CreateImageSourceServiceType`:

https://github.com/dotnet/maui/blob/58d0b1481703850ee71ac01f03c336cef44bf0a4/src/Core/src/Hosting/ImageSources/ImageSourceServiceProvider.cs#L32-L40

I added `UnconditionalSuppressMessage` as I got warnings, but there could be a typo, let me know.

Reply:
I see - this is because you're passing a method with annotation (`CreateImageSourceServiceType` has annotation on one of its parameters) as a delegate somewhere. This means static analysis won't be able to check the actual callsite of the delegate and thus won't be able to validate the annotation on the parameters.

So the suppression is correct, but I would add a different justification and potentially comment explaining why it's needed.